### ISSUE TITLE
Maybe unnecessary rtrim() causing JMailHelper to accept invalid emails

### ISSUE THREAD

**Issue by OctavianC** [joomla/joomla-cms#20075]
### Steps to reproduce the issue
Run code:
```php
// There are dots at the end of the email
var_dump(JMailHelper::isEmailAddress('test@joomla.org....'));
```

### Expected result
bool(false) 

### Actual result
bool(true) 

### System information (as much as possible)
3.8.6

### Additional comments
Any reason why this has rtrim()?
https://github.com/joomla/joomla-cms/blob/staging/libraries/src/Mail/MailHelper.php#L156
I've traced it back to Joomla! 1.5.9 but I don't see it mentioned in the [release notes](https://www.joomla.org/announcements/release-news/5226-joomla-159-security-release-now-available.html)

**Comment by brianteeman** [comment]
You would need to look in the comments for the commit. They are archived here
https://developer.joomla.org/joomlacode-archive/tracker-32.html

**Comment by OctavianC** [comment]
I did take a look but didn't find anything relevant. This personally feels like one of those bugs that was carried over without anyone noticing :) because:

- JFormValidator doesn't accept `test@joomla.org.`:
https://github.com/joomla/joomla-cms/blob/staging/media/system/js/validate-uncompressed.js#L213-L217
eg. try to create a new Joomla! user with that email address - it will fail validation client-side.

- The `EmailRule` class doesn't accept `test@joomla.org.`:
https://github.com/joomla/joomla-cms/blob/staging/libraries/src/Form/Rule/EmailRule.php
To test, run the following:
```php
$rule = JFormHelper::loadRuleType('email');
$field = new SimpleXMLElement('<field></field>');
var_dump($rule->test($field, 'test@joomla.org')); // this works
var_dump($rule->test($field, 'test@joomla.org.')); // this fails
```

Seems like an inconsistency in how `MailHelper::isEmailAddress()` is validating emails.

**Comment by SharkyKZ** [comment]
Issue was related to OpenID:
https://developer.joomla.org/joomlacode-archive/issue-12217.html
http://joomlacode.org/gf/project/joomla/scmsvn/?action=browse&path=%2F&view=rev&revision=11380

Specifically, this part in OpenID plugin:
```
$response->email = str_replace( array('http://', 'https://'), '', $response->username );
$response->email = str_replace( '/', '-', $response->email );
$response->email .= '@openid.';
```

**Comment by OctavianC** [comment]
Thank you so much for digging this up!
Ok, so since this plugin no longer exists I assume I can create a MR to fix this and see if any test units are failing.

**Comment by ghost** [comment]
closed as having Pull Request #20080.
